core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java

```diff
@@ -433,6 +448,43 @@ private Properties tryLoadProperties(
         return ip;
     }
 
+    private static Properties tryLoadCloudProperties(
+            Properties overridProperties, String cloudPropertiesLocations)
+            throws IOException {
+
+        final Properties cp = new Properties();
+        try {
+            String[] locations = cloudPropertiesLocations.split(",");
+            for (String loc : locations) {
+                Path confPath = Paths.get(loc);
+                if (Files.exists(confPath) && Files.isDirectory(confPath)) {
+                    Files.walkFileTree(confPath, new SimpleFileVisitor<Path>() {
+                        @Override
+                        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+                            if (!Files.isDirectory(file)) {
+                                try {
+                                    String val = new String(Files.readAllBytes(file));
+                                    cp.put(file.getFileName().toString(), val);
+                                } catch (IOException e) {
+                                    // Do nothing or should we throw a RuntimeException?
+                                }
+                            }
+                            return FileVisitResult.CONTINUE;
+                        }
+                    });
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        if (overridProperties == null) {
+            return cp;
+        }
+        Properties mergedProperties = new Properties(overridProperties);
```

Comment:
Camel has a Location based properties so we know where they are from that we use for reporting to users. See `OrderedLocationProperties`

Reply:
I'll have a look at this then, thanks.